packages/paypal-commerce-integration/src/paypal-commerce-alternative-methods/paypal-commerce-ratepay-payment-strategy.ts

```diff
@@ -100,42 +110,55 @@ export default class PaypalCommerceRatepayPaymentStrategy implements PaymentStra
-                    set_as_default_stored_instrument: null,
-                    device_info: null,
-                    method_id: payment.methodId,
-                    rate_pay: {
-                        birth_date: this.normalizeDate(ratepayBirthDate),
-                        phone: {
-                            national_number: ratepayPhoneNumber,
-                            country_code: ratepayPhoneCountryCode,
+        return new Promise((resolve, reject) => {
+            const finalizeOrder = async () => {
+                try {
+                    const orderId = await this.paypalCommerceIntegrationService.createOrder(
+                        'paypalcommercealternativemethodscheckout',
+                        { metadataId: this.guid },
+                    );
+
+                    const { ratepayBirthDate, ratepayPhoneNumber, ratepayPhoneCountryCode } =
+                        getFieldsValues();
+
+                    const paymentData = {
+                        formattedPayload: {
+                            vault_payment_instrument: null,
+                            set_as_default_stored_instrument: null,
+                            device_info: null,
+                            method_id: payment.methodId,
+                            rate_pay: {
+                                birth_date: this.normalizeDate(ratepayBirthDate),
+                                phone: {
+                                    national_number: ratepayPhoneNumber,
+                                    country_code: ratepayPhoneCountryCode,
+                                },
+                            },
+                            paypal_account: {
+                                order_id: orderId,
+                            },
                         },
-                    },
-                    paypal_account: {
-                        order_id: orderId,
-                    },
-                },
+                    };
+
+                    await this.paymentIntegrationService.submitOrder(order, options);
+                    await this.paymentIntegrationService.submitPayment({
+                        methodId: payment.methodId,
+                        paymentData,
+                    });
+                    resolve();
+                } catch (error: unknown) {
+                    this.handleError(error);
+                    reject();
+                }
             };
 
-            await this.paymentIntegrationService.submitOrder(order, options);
-            await this.paymentIntegrationService.submitPayment({
-                methodId: payment.methodId,
-                paymentData,
-            });
-        } catch (error: unknown) {
-            this.handleError(error);
-        }
+            this.initializePollingMechanism(
+                payment.methodId,
+                finalizeOrder,
+                payment.gatewayId,
+                this.paypalcommerceratepay,
```

Comment:
it is not necessary to pass `this.paypalcommerceratepay` options because you have an access to it from any method in the class

Reply:
fixed